source/include/core_sntp_client.h

```diff
@@ -0,0 +1,463 @@
+     * send and receive network operations.
+     */
+    UdpTransportIntf_t networkIntf;
+
+    /**
+     * @brief The user-defined interface for incorporating security mechanism of
+     * adding client authentication in SNTP request as well as authenticating server
+     * from SNTP response.
+     *
+     * @note If the application will not use security mechanism for any of the
+     * configured servers, then this interface can be undefined.
+     */
+    SntpAuthenticationIntf_t authIntf;
+
+    /**
+     * @brief Cache of the resolved Ipv4 address of the current server being used for
+     * time synchronization.
+     * As a Best Practice functionality, the client library attempts to resolve the
+     * DNS of the time-server every time the @ref Sntp_SendTimeRequest API is called.
+     */
+    uint32_t currentServerIpV4Addr;
+
+    /**
+     * @brief Cache of the timestamp of sending the last time request to a server
+     * for replay attack protection by checking that the server response contains
+     * the same timestamp in its "originate timestamp" field.
+     */
+    SntpTimestamp_t lastRequestTime;
+
+    /**
+     * @brief State variable for storing the size of the SNTP packet that includes
+     * both #SNTP_PACKET_BASE_SIZE bytes plus any authentication data, if a security
+     * mechanism is used.
+     * This state variable is used for expecting the same size for an SNTP response
+     * from the server.
+     */
+    size_t sntpPacketSize;
+} SntpContext_t;
+
+/**
+ * @brief Initializes a context for SNTP client communication with SNTP/NTP
+ * servers.
+ *
+ * @param[out] pContext The user-supplied memory for the context that will be
+ * initialized to represent an SNTP client.
+ * @param[in] pTimeServers The list of decreasing order of priority of time
+ * servers that should be used by the SNTP client.
+ * @param[in] pNetworkBuffer The user-supplied memory that will be used for
+ * storing network data for SNTP client-server communication. The buffer
+ * MUST stay in scope for all the time the context is in use.
+ * @param[in] bufferSize The size of the passed buffer @p pNetworkBuffer. The buffer
+ * SHOULD be appropriately sized for storing an entire SNTP packet which includes
+ * both #SNTP_PACKET_BASE_SIZE bytes of standard SNTP packet size, and space for
+ * authentication data, if security mechanism is used to communicate with any of
+ * the time servers configured for use.
+ * @param[in] resolveDnsFunc The user-defined function for DNS resolution of time
+ * server.
+ * @param[in] getSystemTimeFunc The user-defined function for querying system
+ * time.
+ * @param[in] setSystemTimeFunc The user-defined function for correcting system
+ * time for every successful time response received from a server.
+ * @param[in] pUdpTransportIntf The user-defined function for performing network
+ * send/recv operations over UDP.
+ * @param[in] pAuthIntf (Optional) The user-defined interface for generating
```

Comment:
Nit: Optional makes it seem like default arguments are supported in C

Reply:
Have removed Optional and added a statement in the description of the parameter to explain it